components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -388,7 +392,52 @@ public Response authorize(@Context HttpServletRequest request, @Context HttpServ
         }
     }
 
+    /**
+     * Add the federated tokens comes with the authentication result to the session data cache.
+     *
+     * @param oAuthMessage         The OAuthMessage with the session data cache entry.
+     * @param authenticationResult The authentication result of authorization call.
+     */
+    private void addFederatedTokensToSessionCache(OAuthMessage oAuthMessage,
+                                                  AuthenticationResult authenticationResult) {
+
+        List<FederatedToken> federatedTokens =
+                (List<FederatedToken>) authenticationResult.getProperty(FEDERATED_TOKENS);
+
+        SessionDataCacheEntry sessionDataCacheEntry = oAuthMessage.getSessionDataCacheEntry();
+        if (sessionDataCacheEntry == null || CollectionUtils.isEmpty(federatedTokens)) {
+            return;
+        }
+        sessionDataCacheEntry.setFederatedTokens(getFederatedTokenDO(federatedTokens));
+        if (log.isDebugEnabled()) {
+            log.debug("Added the federated tokens to the session data cache. Session context identifier: " +
+                    sessionDataCacheEntry.getSessionContextIdentifier());
```

Comment:
IMO, getSessionContextIdentifier is not enough to track along with the flow. can we use authenticatedUser.getLoggableMaskedUserId() from authentication result?

Reply:
Fixed f7da3db475a6513728fdc6a2d49c8e17e3076bab